.github/workflows/push-main-deploy-github-pages.yml

```diff
@@ -25,37 +25,42 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  # Build job
-  build:
```

Comment:
I have removed the comment `# Build job` because we should use self-documenting code where possible.
I've renamed the `build:` job to `build_static_jekyll_site_artifacts:` to ensure it's self-documenting.